src/main/java/ether/TransactionsManager.java

```diff
@@ -0,0 +1,663 @@
+import java.util.concurrent.*;
+import static io.left.rightmesh.mesh.MeshManager.TRANSACTION_RECEIVED;
+
+
+/**
+ * The Transactions manager, responsible to process incoming Ether Transactions from Clients in the Mesh network.
+ */
+public final class TransactionsManager {
+
+    private static final int NUM_OF_THREADS = 100;
+
+    private MeshManager meshManager;
+    private LinkedBlockingQueue<MeshManager.MeshTransactionEvent> transactionsQueue = new LinkedBlockingQueue<>();
+    private ConcurrentHashMap<String, EtherUtility.PaymentChannel> channelsMap = new ConcurrentHashMap<>();
+    private ExecutorService executor = Executors.newFixedThreadPool(NUM_OF_THREADS);
+    private Thread queueThread = null;
+    private volatile boolean isRunning = false;
+    private Http httpAgent;
+    private MeshID ownMeshId;
+
+
+    private static volatile TransactionsManager instance = null;
+
+    public static TransactionsManager getInstance(MeshManager mm) {
+        if (instance == null) {
+            synchronized(TransactionsManager.class) {
+                if (instance == null) {
+                    instance = new TransactionsManager(mm);
+                }
+            }
+        }
+
+        return instance;
+    }
+
+    private TransactionsManager(MeshManager mm){
+        httpAgent = new Http(Settings.RPC_ADDRESS, Settings.DEBUG_INFO);
+        meshManager = mm;
+        ownMeshId = mm.getUuid();
+        mm.on(TRANSACTION_RECEIVED, this::handleTransactionPacket);
+    }
+
+
+    /**
+     * Handles Transaction packets from the Mesh network.
+     * @param rmEvent
+     */
+    private void handleTransactionPacket(MeshManager.RightMeshEvent rmEvent) {
+        if(Settings.DEBUG_INFO) {
+            System.out.println("Transaction received.");
+        }
+
+        MeshManager.MeshTransactionEvent event = (MeshManager.MeshTransactionEvent)rmEvent;
+        insertTransaction(event);
+    }
+
+    /**
+     * Returns Running state of the Transaction Manager
+     * @return
+     */
+    public synchronized boolean isRunning(){
+        return isRunning;
+    }
+
+
+    /**
+     * Starts the Transactions Manager
+     */
+    public synchronized void start(){
+        if(!isRunning){
+            queueThread = new Thread(this::processTransactionsQueue);
+            queueThread.start();
+            isRunning = true;
+        }
+    }
+
+
+    /**
+     * Stops the Transactions Manager and cleans resources.
+     */
+    public synchronized void stop(){
+        isRunning = false;
+        insertStopMessage();
+
+        if(queueThread != null && queueThread.isAlive()) {
+            try {
+                queueThread.join();
+            } catch (InterruptedException e) {
+                if (Settings.DEBUG_INFO) {
+                    System.out.println("InterruptedException: " + e);
+                }
+            }
+        }
+
+        try {
+            executor.shutdown();
+            executor.awaitTermination(5, TimeUnit.SECONDS);
```

Comment:
Should this be configured somewhere? Or at least moved to a constant/documented why 5 seconds?